components/auth/org.wso2.carbon.auth.scim.rest.api/src/main/java/org/wso2/carbon/auth/scim/rest/api/impl/UsersApiServiceImpl.java

```diff
@@ -1,57 +1,107 @@
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
+
     @Override
-    public Response usersIdGet(String id
-  ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response usersIdGet(String id, Request request) throws NotFoundException {
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            UserResourceManager userResourceManager = new UserResourceManager();
+            SCIMResponse scimResponse = userResourceManager.get(id, userManager, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
+
     @Override
-    public Response usersIdPut(String id, UserDTO body
-  ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response usersIdPut(String id, UserDTO body, Request request) throws NotFoundException {
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            UserResourceManager userResourceManager = new UserResourceManager();
+            SCIMResponse scimResponse = userResourceManager.updateWithPUT(id, body.toString(), userManager, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
+
     @Override
-    public Response usersPost(UserDTO body
-  ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response usersPost(UserDTO body, Request request) throws NotFoundException {
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            UserResourceManager userResourceManager = new UserResourceManager();
+            SCIMResponse scimResponse = userResourceManager.create(body.toString(), userManager, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
```

Comment:
Exception has not been logged. Need to log exception as follows,

log.error("Message", e);